src/porepy/io/grid_writer.py

```diff
@@ -0,0 +1,484 @@
+    else:
+        addCellFaceTagGrid(gb, tol)
+
+
+def addCellFaceTagGrid(g, tol=1e-10):
+
+    g.cell_facetag = np.zeros(g.cell_faces.indptr[-1], dtype=int)
+    if g.dim == 3:
+        faces_per_cell = 6
+    elif g.dim == 2:
+        faces_per_cell = 4
+    elif g.dim < 2:
+        return
+    else:
+        raise ValueError(
+            "Invalid grid dimension (must be 0,1,2, or 3): {}".format(g.dim)
+        )
+
+    if g.dim <= 2:
+        R = pp.map_geometry.project_plane_matrix(g.nodes, reference=np.array([0, 1, 0]))
+        cell_centers, _, face_centers, _, _, _ = pp.map_geometry.map_grid(g, R=R)
+    else:
+        cell_centers = g.cell_centers
+        face_centers = g.face_centers
+    if not ("CartGrid" in g.name or "TensorGrid" in g.name):
+        raise ValueError("Can only enforce face ordering for CartGrid or TensorGrid")
+
+    for k in range(g.num_cells):
+        cc = cell_centers[:, k]
+        for i in range(faces_per_cell):
+            face = g.cell_faces.indices[k * faces_per_cell + i]
+            fc = face_centers[:, face]
+
+            diff = cc - fc
+            num_tags = 0
+            if diff[0] > tol:
+                g.cell_facetag[k * faces_per_cell + i] = 0
+                num_tags += 1
+            if diff[0] < -tol:
+                g.cell_facetag[k * faces_per_cell + i] = 1
+                num_tags += 1
+            if diff[1] > tol:
+                g.cell_facetag[k * faces_per_cell + i] = 2
+                num_tags += 1
+            if diff[1] < -tol:
+                g.cell_facetag[k * faces_per_cell + i] = 3
+                num_tags += 1
+            if g.dim == 3 and diff[2] > tol:
+                g.cell_facetag[k * faces_per_cell + i] = 4
+                num_tags += 1
+            if g.dim == 3 and diff[2] < -tol:
+                g.cell_facetag[k * faces_per_cell + i] = 5
+                num_tags += 1
+
+            if num_tags != 1:
+                raise AttributeError(
+                    "Could not find W, E, S, or N face of cell {}".format(k)
+                )
+
+
+def enforce_opm_face_ordering(gb):
+    if isinstance(gb, pp.GridBucket):
+        for g, _ in gb:
+            enforce_opm_face_ordering_grid(g)
+    else:
+        enforce_opm_face_ordering_grid(gb)
+
+
+def enforce_opm_face_ordering_grid(g):
+    # OPM faces ordered counterclockwise starting at West face
+    if g.dim == 2:
+        opm_sort = [0, 2, 1, 3]
+    elif g.dim == 3:
+        opm_sort = [0, 1, 2, 3, 4, 5]
+    else:
+        raise ValueError(
+            "Can not assign face ordering for grids of dimension {}".format(g.dim)
+        )
+
+    if not hasattr(g, "cell_facetag"):
+        raise ValueError("Can only order grids with cell_facetag")
+
+    # Get ordering of OPM faces in a cell
+    _, IC = np.unique(opm_sort, return_inverse=True)
+    cell_facetag = g.cell_facetag.reshape((-1, len(opm_sort)))
+
+    old2new = np.empty(g.num_faces, dtype=int)
+    for k in range(g.num_cells):
```

Comment:
I was a bit worried that the reordering may be problematic for other parts of the code, but it should be okay to reorder faces. Nodes in the face-node relation would have been another matter.